connection.go

```diff
@@ -33,17 +33,35 @@ func (c *conn) ExecContext(ctx context.Context, query string, args []driver.Name
 		panic("database/sql/driver: misuse of duckdb driver: ExecContext after Close")
 	}
 
-	if len(args) == 0 {
-		// This should be removed once duckdb_extract_statements and related APIs are available to parse multiple statements
-		// so query cancellation works for this case as well
-		return c.execUnprepared(query)
+	stmts, size, err := c.extractStmts(query)
+	if err != nil {
+		return nil, err
 	}
 
-	stmt, err := c.prepareStmt(query)
+	// execute all statements but the last one, apply args only to last statement and return result of last statement
+	for i := C.idx_t(0); i < size-1; i++ {
+		stmt, err := c.prepareExtractedStmt(stmts, i)
+		if err != nil {
+			return nil, err
+		}
+		// send nil args to execute statement and ignore result
+		_, err = stmt.ExecContext(ctx, nil)
+		if err != nil {
+			return nil, err
+		}
+		err = stmt.Close()
+		if err != nil {
+			return nil, err
+		}
```

Comment:
This will not close the statement if `ExecContext` errors

Reply:
should we just ignore the error of `stmt.Close()` as even with defer we would have ignored it, it would have been printed though I think.